runtime/compiler/ilgen/Walker.cpp

```diff
@@ -7165,6 +7165,13 @@ TR_J9ByteCodeIlGenerator::storeAuto(TR::DataType type, int32_t slot, bool isAdju
       }
 
    symRef = symRefTab()->findOrCreateAutoSymbol(_methodSymbol, slot, type, true, false, true, isAdjunct);
+
+   // Self-storing, skip the store
+   if (storeValue->getOpCode().isLoadDirect() && storeValue->getOpCode().hasSymbolReference() && storeValue->getSymbolReference() == symRef)
```

Comment:
do we have to care about direct rdbars if one was generated? or if we are supposed to generate a wrbar direct?

Reply:
rdbar and wrbar are not used in auto load/store

Reply:
well the opcodes exist, but I guess we can skip for now since we don't use them in OpenJ9 which is a fair point.